listenbrainz/webserver/static/js/src/Loader.tsx

```diff
@@ -0,0 +1,25 @@
+/* eslint-disable react/prefer-stateless-function */
+import * as React from "react";
+import { faSpinner } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { IconProp } from "@fortawesome/fontawesome-svg-core";
+
+type withLoaderProps = {
+  isLoading: boolean;
+};
+
+const withLoader = <P extends object>(Component: React.ComponentType<P>) => {
+  return class WithLoader extends React.Component<P & withLoaderProps> {
+    render() {
+      const { isLoading, ...props } = this.props;
+      // TODO: Replace with Bootstrap 4 spinner when we upgrade
+      return isLoading ? (
+        <FontAwesomeIcon icon={faSpinner as IconProp} size="4x" spin />
+      ) : (
+        <Component {...(props as P)} />
```

Comment:
I would do this using `{props.children}` here instead of `<Component … />`.
Then you'll be able to simply invoke your component with:
```
<Loader isLoading={loading}>
    <Bar data={data} maxValue={maxListens} />
</Loader>
```
And your Loader component can be a regular React.Component class like the others instead of a factory, as an added bonus.

Here's a good breakdown of props.children in React: https://codeburst.io/a-quick-intro-to-reacts-props-children-cb3d2fce4891

Reply:
Yes, this is simpler, should have thought about it before :D